uvcombine/uvcombine.py

```diff
@@ -1002,39 +1075,110 @@ def feather_simple_cube(cube_hi, cube_lo,
 
     if cube_lo.shape[0]!=cube_hi.shape[0] or not all(cube_lo.spectral_axis == cube_hi.spectral_axis):
         if allow_spectral_resample:
-            cube_lo = cube_lo.spectral_interpolate(cube_hi.spectral_axis)
+            cube_lo = cube_lo.spectral_interpolate(cube_hi.spectral_axis, **save_kwargs)
         else:
             raise ValueError("Spectral axes do not match. Enable `allow_spectrum_resample` to "
                              "spectrally match the low resolution to high resolution data.")
 
-    feath_array = np.empty(cube_hi.shape)
+    # If cubes are DaskSpectralCubes, use the dask implementation
+    if isinstance(cube_hi, DaskSpectralCube) and isinstance(cube_lo, DaskSpectralCube):
+
+        # The block mapping has to be the same. Set here whether to
+        # allow a prior reproject operation for the SD to match.
+        if allow_lo_reproj:
+            # NOTE: is this memory friendly? We don't have a dedicated
+            # dask reprojection task, so this COULD break things.
+            cube_lo = cube_lo.rechunk(('auto', -1, -1), **save_kwargs)
+            cube_lo_reproj = cube_lo.reproject(cube_hi.header, use_memmap=use_memmap)
+        else:
+            cube_lo_reproj = cube_lo
+
+        # Check that the pixel sizes of both cubes now match
+        equal_sizes = cube_lo_reproj.shape == cube_hi.shape
+        if not equal_sizes:
+            raise ValueError("The cube_lo array shape does not match the cube_hi"
+                             " shape. Enable `allow_lo_reproj` or reproject cube_lo"
+                             " before feathering.")
+
+        # Ensure spatial chunk sizes are matched.
+        if force_spatial_rechunk:
+            cube_hi = cube_hi.rechunk(('auto', -1, -1), **save_kwargs)
+            cube_lo = cube_lo.rechunk(('auto', -1, -1), **save_kwargs)
+
+            if cube_hi._data.chunksize != cube_lo._data.chunksize:
+                raise ValueError("The chunk size does not match between the cubes."
+                                 f" cube_hi: {cube_hi._data.chunksize} "
+                                 f" cube_lo: {cube_lo._data.chunksize} "
+                                 "Check reprojection or apply prior to feathering.")
+
+        # Check kwargs for feather_simple kwarg to allow matching units
+        match_units = kwargs.pop('match_units', True)
+
+        # Check for units consistency
+        if match_units:
+            # After this step, the units of cube_hi are some sort of surface brightness
+            # unit equivalent to that specified in the high-resolution header's units
+
+            # NOTE: is this memory-friendly??
+            cube_lo = cube_lo.to(cube_hi.unit)
+
+            # When in a per-beam unit, we need to scale the low res to the
+            # Jy / beam for the HIRES beam.
+            jybm_unit = u.Jy / u.beam
+            if cube_hi.unit.is_equivalent(jybm_unit):
+                cube_lo *= (cube_hi.beam.sr / cube_lo.beam.sr).decompose().value
```

Comment:
ditto
```suggestion
            if cube_hi.unit.is_equivalent(jybm_unit):
                cube_lo_reproj *= (cube_hi.beam.sr / cube_lo_reproj.beam.sr).decompose().value
```
this is important also b/c `*=` could look like it was going to modify user input if this weren't already a copy